src/app/utilities/file-import-helper.ts

```diff
@@ -0,0 +1,37 @@
+import {MessageHelper} from './message-helper';
+
+export class FileImportHelper {
+
+  static importCriteria(elementId: string, reader: FileReader, isUploadListenerNotAdded: boolean) {
+    let uploadElm = document.getElementById(elementId);
+    if (isUploadListenerNotAdded) {
+      uploadElm.addEventListener('change', (event) => this.fileUpload(event, reader), false);
+    }
+    // reset the input path so that it will take the same file again
+    uploadElm['value'] = '';
+    uploadElm.click();
+  }
+
+  static fileUpload(event, reader: FileReader) {
+    MessageHelper.alert('info', 'File is being processed, waiting for response.');
+    let file = event.target.files[0];
+    reader.readAsText(file);
```

Comment:
also a message telling it's finished?

Reply:
I can add one, but there are already 3 messages showing up, when successfully loading file - 1 on start and 2 when executing restoreCohort. Maybe it would be too much?

Reply:
I would put up a message at the beginning of the import (inside the importCriteria method of gb-cohort-selection.component.ts) saying it has started, and another message when onLoad is finished, i.e. in handleVariablesFileUploadEvent saying it is finished, and in case error, say there is error (which we already have)